workers/proc.sign.eos.ext.wrk.js

```diff
@@ -0,0 +1,286 @@
+          throw new Error('STATE_TABLE_INVALID_DATA')
+        }
+
+        return res
+      }],
+
+      pendingRelDone: ['lastIrreversibleBlockTimes', 'pendingLocal', 'stateRemote', async (res) => {
+        const rTime = getTimestamp(
+          res.lastIrreversibleBlockTimes.remote.last_irreversible_block_timestamp
+        )
+        const data = await remoteRpc.getTableRows({ table: TABLE_RELEASE_DONE })
+
+        if (JSON.stringify(state[remote].preldone) !== JSON.stringify(data.rows)) {
+          state[remote].preldone = data.rows
+          console.log(remote, 'pendingRelDone', JSON.stringify(data.rows))
+        }
+
+        let nextRelDoneId = 0
+
+        // the `releasedone` action removes all rows from the `pndreldones` with
+        // ids less than or equal to the id sent in the `releasedone` command
+        // so we just end the iteration with the highest id found
+        _.forEach(data.rows, (entry) => {
+          const { id, time } = entry
+          const tMil = time * 1000
+
+          console.log('rtime', rTime, tMil)
+          if (rTime < tMil) {
+            console.log('pendingRelDone abort:', id, rTime, '<', time)
+            return false
+          }
+
+          nextRelDoneId = id
+        })
+
+        // also skip nextRelDoneId 0 (not found)
+        if (!nextRelDoneId) {
+          return { lastDoneReleaseDoneId: 0 }
+        }
+
+        await this.sendTx(remote, 'releasedone', { id: nextRelDoneId })
+
+        return { lastDoneReleaseDoneId: nextRelDoneId }
+      }],
+
+      processPendingTransfers: ['pendingRelDone', (res, next) => {
+        const lTime = getTimestamp(
+          res.lastIrreversibleBlockTimes.local.last_irreversible_block_timestamp
+        )
+
+        console.log(local, 'processPendingTransfers', JSON.stringify(res))
+        console.log('processing', res.pendingLocal)
+
+        let { nextTransId, lastDoneReleaseDoneId } = res.stateRemote
+
+        const pendingTransfers = res.pendingLocal
+        async.eachSeries(pendingTransfers, async (entry) => {
+          console.log('---> nextTransId, lastDoneReleaseDoneId', nextTransId, lastDoneReleaseDoneId)
+          const { id, time, account, quantity, memo } = entry
+          const tMil = time * 1000
+
+          console.log('ltime', lTime, tMil)
+          if (lTime < tMil) {
+            console.log('pendingTransfers abort:', id, lTime, '<', time)
+            throw new ProcessingError('ERR_PND_TRANSFERS')
+          }
+
+          if (id < nextTransId) {
+            if (id <= lastDoneReleaseDoneId) {
+              console.log('confirming transfer id:', id)
+              await this.sendTx(local, 'transferdone', { id, account, quantity })
+
+              return
+            }
+          }
+
+          if (id === nextTransId) {
+            console.log('releasing id:', id)
+            await this.sendTx(remote, 'release', { id, account, quantity, memo })
+
+            nextTransId = nextTransId + 1
```

Comment:
potential bottleneck: we depend on local state here, which means we have to use `async.eachSeries` for now, which is limited to one async operation at the time.